Library/Homebrew/tap.rb

```diff
@@ -280,4 +316,93 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreFormulaTap < Tap
```

Comment:
> We can replace them to symlinks.

:+1: 



> Sadly we can't due to current update method. One goal I want to ensure is that brew update should pass on all of existed Travis CI OS X images.

It's definitely possible with the current update method; it'd just require some smart use of `git` to ensure it plays nicely.

> This certainly can be achieved, but it's more sense to be handled internally by maintainers rather than average users.

I'm not sure I agree here; I've seen various use-cases in the past and had people ask for this feature (I don't want Homebrew's stuff, I want to ensure the versions never change unless I want them to). It'd be an "Advanced User" feature for sure and perhaps just something that we make it relatively easy to e.g. run Homebrew as a fork to do (like Linuxbrew).